include/onnxruntime/interface/framework/kernel.h

```diff
@@ -0,0 +1,227 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <cassert>
+#include <numeric>
+#include "interface/common/data_types.h"
+#include "core/common/status.h"
+
+namespace onnxruntime {
+
+namespace interface {
+
+using TensorShape = std::vector<int64_t>;
+
+struct IKernelContext {
+  virtual ~IKernelContext() = default;
+  virtual const void* InputData(int index) const = 0;
```

Comment:
>void* InputData(int index)  [](http://example.com/codeflow?start=16&length=27)

the return value is a MLValue? or the tensor's data buffer?

Reply:
We can return IArg for sure.